src/main/java/pro/trevor/tankgame/state/board/unit/GenericTank.java

```diff
@@ -4,23 +4,22 @@
 import pro.trevor.tankgame.rule.type.IPlayerElement;
 import pro.trevor.tankgame.rule.type.ITickElement;
 import pro.trevor.tankgame.state.board.GenericElement;
-import pro.trevor.tankgame.state.board.attribute.IAttribute;
 import pro.trevor.tankgame.state.board.IMovable;
 import pro.trevor.tankgame.state.board.Position;
 
 import java.util.*;
 
-public class GenericTank<E extends Enum<E> & IAttribute> extends GenericElement<E> implements IMovable, ITickElement, IPlayerElement, IUnit {
+public abstract class GenericTank extends GenericElement implements IMovable, ITickElement, IPlayerElement, IUnit {
```

Comment:
This shouldn't be abstract at some point v3 should directly use this

Reply:
ACK